### DISCUSSION TITLE
Rotation of single segments

### DISCUSSION THREAD

**Discussion by LeonardoGentile** [mifi/lossless-cut#2026] [Support]
Hello, 
Is it possible to apply different rotations to different segments of the same original input video? 
I was able to apply the general rotation for the entire output but not for single segments. Maybe I couldn't find that options? 🤷‍♂️

**Comment by mifi** [comment]
https://github.com/mifi/lossless-cut/blob/master/issues.md#rotation-and-merging

**Reply by LeonardoGentile** [reply]
ok thanks, but I've encounter a weird issue when rotating single segments with external tools. 
I've used Lossless Cut to cut several segments from the same video, I then rotated some of them in one direction and others in another direction using the MacOs integrated Quicklook tool. If I open these videos again with quicklook, quicktime, IINA or VLC they are played correctly showing me the correct rotation (the one I've applied). 
But If I (batch) merge back all of these segments using Lossless cut then my rotations are not taken into consideration, they still have the orientation of the original video they came from. 
I've tried to click on use metadata options but it doesn't solve the problem

**Reply by mifi** [reply]
that's because it's not possible to have multiple different rotation values for different segments when merging, see https://github.com/mifi/lossless-cut/blob/master/issues.md#rotation-and-merging

**Reply by arpcpro** [reply]
I’m experiencing the same issue on the Windows version. I have *.mp4 H.265 segments recorded on a Samsung phone where some clips are upside down (180°), and others are correct.

When merging them, LosslessCut v3.67.2 ignores the individual rotation flags. Even worse, I tried using ExifTool to "fix" the orientation flags of the segments before importing them into LosslessCut, but LosslessCut completely ignores those updated flags during the merge process. It seems to only look at the original video metadata, resulting in a final file where these specific segments remain upside down.

This essentially breaks the "lossless" workflow for anyone using modern smartphones that rely on rotation metadata. We need LosslessCut to **honor the rotation metadata of each segment during the concat process.**
This without forcing a full re-encode of the pixel data.

It's frustrating because the metadata is there, but the merge function is stripping or ignoring it, forcing us back to destructive re-encoding.

**Reply by mifi** [reply]
@arpcpro I think you are misunderstanding here. File format's don't have a way to change rotation during the course of playback. So what you're asking for is **technically impossible. If you merge files that are recorded with different orientations, you can only keep a single orientation in the output file**

**Reply by arpcpro** [reply]
@mifi Thanks for the reply. You're totally right. I understand that the video containers only supports a single global rotation flag. Being so, to solve this within the LosslessCut workflow, the app would need to implement a "Smart Merge with transcoding" feature. If LosslessCut detects conflicting rotation flags during a merge, it could offer to transcode only the specific segments that don't match the primary orientation (e.g., re-encoding the 180° clips to match) before performing the lossless concat with the rest. This would allow the majority of the video to remain untouched while fixing the metadata mismatch. Something to add to the feature request list I guess.

**Reply by mifi** [reply]
That would be outside the current scope of LosslessCut as it requires lossy operations #372
